Firestore/core/src/firebase/firestore/nanopb/writer.h

```diff
@@ -144,6 +145,17 @@ class StringWriter : public Writer {
   std::string buffer_;
 };
 
+/** Creates a `Writer` that writes into a `grpc::ByteBuffer`. */
+class GrpcByteBufferWriter : public Writer {
```

Comment:
As with discussion of `Reader`, I think this could likely go in the `remote` directory, alongside all the rest of our grpc-related stuff.